datajunction-server/datajunction_server/api/helpers.py

```diff
@@ -358,19 +358,16 @@ def validate_node_data(
         raise DJException(message=str(raised_exceptions)) from raised_exceptions
 
     # Only raise on missing parents if the node mode is set to published
-    if missing_parents_map:
-        if validated_node.mode == NodeMode.DRAFT:
-            validated_node.status = NodeStatus.INVALID
-        else:
-            raise DJException(
-                errors=[
-                    DJError(
-                        code=ErrorCode.MISSING_PARENT,
-                        message="Node definition contains references to nodes that do not exist",
-                        debug={"missing_parents": list(missing_parents_map.keys())},
-                    ),
-                ],
-            )
+    if missing_parents_map and validated_node.mode != NodeMode.DRAFT:
+        raise DJException(
+            errors=[
```

Comment:
I know this wasn't created in this PR but this exception should include `http_status_code=HTTPStatus.BAD_REQUEST`